src/app/common/utils/helpers/inheritance-resolver.js

```diff
@@ -20,49 +19,39 @@ export default class InheritanceResolver {
     return this.cachedDataStructures.size;
   }
 
-  getCachedDataStructure(member) {
-    const cacheKey = this.getMemberCacheKey(member);
-    const cachedDataStructure = this.cachedDataStructures.get(cacheKey);
-
-    if (!cachedDataStructure) return null;
+  canUseCache(memberSource) {
+    const hasContentOverrided = !!memberSource.content;
 
-    const canUseFromCache = this.canUseFromCache(member);
-
-    return canUseFromCache ? cachedDataStructure : null;
+    return !hasContentOverrided;
   }
 
-  canUseFromCache(member) {
-    const referenceDataStructure = member.referenceDataStructure || member.element;
-
-    if (!referenceDataStructure) return false;
+  getCachedDataStructure(member) {
+    const canUseFromCache = this.canUseCache(member);
 
-    const referencedDS = this.categories.dataStructuresArray.find(ds => ds.id === referenceDataStructure);
-    const refDSContent = get('content', 'content').from(referencedDS);
+    if (!canUseFromCache) return null;
 
-    if (referenceDataStructure === member.element) return !member.content;
+    const cacheKey = this.getCacheKey(member);
+    const cachedDataStructure = this.cachedDataStructures.get(cacheKey) || null;
```

Comment:
maybe we can just return `this.cachedDataStructures.get(cacheKey)`? I suppose there is no difference whether we return `null` or `undefined`

Reply:
ok, done